### PR TITLE
Fix `<ReferenceInput>` throws a recoverable error in production

### PR DESCRIPTION
**Pull request by fzaninotto**
## Problem

Since #9637, users see an error in production while the app isn't broken

## Solution

Only throw in development.

### CODE REVIEW

**Review by adguernier — APPROVED** [review]

**Review by digilist — APPROVED** [review]

File: packages/ra-ui-materialui/src/input/ReferenceInput.tsx [hunk 1]
```diff
@@ -84,14 +84,11 @@ export const ReferenceInput = (props: ReferenceInputProps) => {
...
 
-    if (props.validate) {
+    if (props.validate && process.env.NODE_ENV !== 'production') {
         throw new Error(
             '<ReferenceInput> does not accept a validate prop. Set the validate prop on the child instead.'
         );
     }
-    if (Children.count(children) !== 1) {
-        throw new Error('<ReferenceInput> only accepts a single child');
-    }
```
Comment (slax57) [inline packages/ra-ui-materialui/src/input/ReferenceInput.tsx:94]: **Question:** Why did you remove this test?

Also, this creates a linter warning at line 1.

**Review by slax57 — CHANGES_REQUESTED** [review]

File: packages/ra-ui-materialui/src/input/ReferenceInput.tsx (hunk 1 above)
Comment (fzaninotto) [reply packages/ra-ui-materialui/src/input/ReferenceInput.tsx:94]: I removed it because it was only required when we cloned the child. We no longer do that, so there is no need to force a single child.

Comment (fzaninotto) [reply packages/ra-ui-materialui/src/input/ReferenceInput.tsx:94]: Also, I fixed the warning

**Review by slax57 — APPROVED** [review]
